packages/grafana-ui/src/components/RefreshPicker/RefreshPicker.tsx

```diff
@@ -57,18 +69,43 @@ export class RefreshPicker extends PureComponent<Props> {
   }
 
   render() {
-    const { onRefresh, intervals, tooltip, value, text, isLoading, noIntervalPicker, width } = this.props;
+    const {
+      onRefresh,
+      intervals,
+      tooltip,
+      value,
+      text,
+      isLoading,
+      noIntervalPicker,
+      width,
+      offOptionLabelMsg,
+      offOptionAriaLabelMsg,
+      offDescriptionAriaLabelMsg,
+      onDescriptionAriaLabelMsg,
+    } = this.props;
 
     const currentValue = value || '';
     const variant = this.getVariant();
-    const options = intervalsToOptions({ intervals });
+    const translatedOffOption = {
+      value: RefreshPicker.offOption.value,
+      label: offOptionLabelMsg || RefreshPicker.offOption.label,
+      ariaLabel: offOptionAriaLabelMsg || RefreshPicker.offOption.ariaLabel,
+    };
+    const options = intervalsToOptions({ intervals, offOption: translatedOffOption });
     const option = options.find(({ value }) => value === currentValue);
-    let selectedValue = option || RefreshPicker.offOption;
+    let selectedValue = option || translatedOffOption;
```

Comment:
Passing the `translatedOffOption` to `intervalsToOptions()` and then assigning it here manually is a bit confusing after `option` being assigned one of the `options` which now have `translatedOffOption` with them. Since `currentValue` can be `''` and `translatedOffOption.value` is always `''`, it seems we don't need this `selectedValue` as a separate variable at all?

Reply:
I believe the reason `selectedValue` is there, is for the case where we did not find any `option` that matches the value, then we default it to the `offOption`. We could make the change to just have:
`let selectedValue = options.find(({ value }) => value === currentValue) || translatedOffOption;`
But we don't gain that much by it. what do you think?

Reply:
We are always adding the `offOption` to the `options` array before digging into it, right here: `options.unshift(offOption)`. That's why I think `option` has the ability to be the `translatedOffOption` when there is no `value` passed to RefreshPicker and it is assigned the empty string here: `const currentValue = value || ''`. It seems to me option will always be defined, but maybe I'm missing something?

Reply:
The case that I mentioned before is not for when no `value` has been passed but more what if the `value` doesn't match any of the `options`? If we pass in `value = '50s'` for example, that is not in the possible intervals so `option` will be `undefined` and so we need to have a backup for that case and that is where `selectedValue` comes in. Maybe I'm not explaining it correctly 😅

Reply:
Аааааh yes, of course! Thanks for the clarification. You are right.